Branch-SDK/src/io/branch/referral/ServerResponse.java

```diff
@@ -135,4 +140,28 @@ public JSONArray getArray() {
 	public BranchLinkData getLinkData() {
 		return linkData_;
 	}
+
+	/**
+	 * <p>Sets the request object caused this response, as a generic {@link Object}
+	 * instance. This object can be type-cast by other methods within this class:</p>
+	 *
+	 * @param request Generic {@link Object} instance containing the request data caused this response.
+	 */
+	public void setRequestObject(Object request) {
+		this.requestObject_ = request;
+	}
+
+	/**
+	 * <p>Checks whether the request params cause this response is an instance of a
+	 * {@link JSONObject} object, and if so type-casts it to the corresponding class type.</p>
+	 *
+	 * @return A {@link JSONObject} containing the post params sent with corresponding  server request, or a null
+	 * value if the post params is not of the {@link JSONObject} type.
+	 */
+	public JSONObject getRequestObject() {
```

Comment:
Yeah, that's exactly how it should be. That should make it a whole lot easier to deal with, without having to worry about any sort of thread management or trying to drag items around through different scopes. Everything just stays where it belongs.